algorithms/src/std_algorithms/impl/Kokkos_RandomAccessIterator.hpp

```diff
@@ -164,6 +163,10 @@ class RandomAccessIterator<::Kokkos::View<DataType, Args...>> {
     return it;
   }
 
+  friend iterator_type operator+(difference_type n, iterator_type other) {
+    return other + n;
+  }
```

Comment:
This is the relevant change to make the class fulfill the std::random_access_iterator concept. Thhe rest of the changes are for allowing `const View`s.

Reply:
(other changes were dropped)